Couple of minor cleanups

bjorn3 opened the pull request.

bjorn3 committed:
Include allocator module in LLVM_passes timer

bjorn3 committed:
Remove unused field from ModuleConfig

bjorn3 commented:
@bors r=WaffleLapkin

bors commented:
:pushpin: Commit 6fbea4ffe84133edf74a0069ffd18f982527e53c has been approved by `WaffleLapkin`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=WaffleLapkin 6fbea4ffe84133edf74a0069ffd18f982527e53c -->
<!-- homu: {"type":"Approved","sha":"6fbea4ffe84133edf74a0069ffd18f982527e53c","approver":"WaffleLapkin","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bjorn3 commented:
cc https://github.com/rust-lang/rust/pull/117364 which removed embedding of LLVM bitcode on Apple platforms.

rust-log-analyzer commented:
The job **`x86_64-apple-2`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/16818963644/job/47642017314) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/47642017314) [(plain)](https://github.com/rust-lang/rust/commit/0203e7958516c0a1be459ad5058be4b733604f6b/checks/47642017314/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
---- [ui] tests/ui/linking/ld64-cross-compilation.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/Users/runner/work/rust/rust/build/x86_64-apple-darwin/stage2/bin/rustc" "/Users/runner/work/rust/rust/tests/ui/linking/ld64-cross-compilation.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/Users/runner/.cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/Users/runner/work/rust/rust/vendor" "--sysroot" "/Users/runner/work/rust/rust/build/x86_64-apple-darwin/stage2" "--target=x86_64-apple-darwin" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/Users/runner/work/rust/rust/build/x86_64-apple-darwin/test/ui/linking/ld64-cross-compilation/a" "-A" "internal_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/Users/runner/work/rust/rust/build/x86_64-apple-darwin/native/rust-test-helpers" "-Copt-level=3" "-Ccodegen-units=256" "-Clink-arg=-ld_classic"
stdout: none
--- stderr -------------------------------
error: linking with `cc` failed: exit status: 1
   |
   = note:  "cc" "/var/folders/vk/nx37ffx50hv5djclhltc26vw0000gn/T/rustcmJ38sD/symbols.o" "<5 object files omitted>" "<sysroot>/lib/rustlib/x86_64-apple-darwin/lib/libstd-9ba57eaadfeae4a1.dylib" "<sysroot>/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-*.rlib" "-lSystem" "-lc" "-lm" "-arch" "x86_64" "-mmacosx-version-min=10.12.0" "-L" "/Users/runner/work/rust/rust/build/x86_64-apple-darwin/native/rust-test-helpers" "-o" "/Users/runner/work/rust/rust/build/x86_64-apple-darwin/test/ui/linking/ld64-cross-compilation/a" "-Wl,-dead_strip" "-nodefaultlibs" "-Wl,-rpath,@loader_path/../../../../stage2/lib/rustlib/x86_64-apple-darwin/lib" "-ld_classic"
   = note: some arguments are omitted. use `--verbose` to show all linker arguments
   = note: ld: in /Users/runner/work/rust/rust/build/x86_64-apple-darwin/stage2/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-5de5c4c90ca8f717.rlib(compiler_builtins-5de5c4c90ca8f717.compiler_builtins.abc1b0ccfdbf58c6-cgu.000.rcgu.o), Object file with bitcode missing cmdline options: /Users/runner/work/rust/rust/build/x86_64-apple-darwin/stage2/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-5de5c4c90ca8f717.rlib(compiler_builtins-5de5c4c90ca8f717.compiler_builtins.abc1b0ccfdbf58c6-cgu.000.rcgu.o) file '/Users/runner/work/rust/rust/build/x86_64-apple-darwin/stage2/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-5de5c4c90ca8f717.rlib'
           clang: error: linker command failed with exit code 1 (use -v to see invocation)
           

error: aborting due to 1 previous error
------------------------------------------

```

</details>

bjorn3 committed:
Remove bitcode_llvm_cmdline

It used to be necessary on Apple platforms to ship with the App Store,
but XCode 15 has stopped embedding LLVM bitcode and the App Store no
longer accepts apps with bitcode embedded.

bjorn3 commented:
On macOS the linker rejects object files with bitcode but missing cmdline. I changed it to keep the cmdline on all platforms, just always empty.

```
= note: ld: in /Users/runner/work/rust/rust/build/x86_64-apple-darwin/stage2/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-5de5c4c90ca8f717.rlib(compiler_builtins-5de5c4c90ca8f717.compiler_builtins.abc1b0ccfdbf58c6-cgu.000.rcgu.o), Object file with bitcode missing cmdline options: /Users/runner/work/rust/rust/build/x86_64-apple-darwin/stage2/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-5de5c4c90ca8f717.rlib(compiler_builtins-5de5c4c90ca8f717.compiler_builtins.abc1b0ccfdbf58c6-cgu.000.rcgu.o) file '/Users/runner/work/rust/rust/build/x86_64-apple-darwin/stage2/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-5de5c4c90ca8f717.rlib'
```

WaffleLapkin commented:
@bors r+
